iohk-monitoring/src/Cardano/BM/Data/Trace.lhs

```diff
@@ -5,23 +5,22 @@
 %if style == newcode
 \begin{code}
 {-# LANGUAGE RankNTypes          #-}
-{-# LANGUAGE LambdaCase          #-}
 {-# LANGUAGE ScopedTypeVariables #-}
 
 module Cardano.BM.Data.Trace
   ( Trace
   )
   where
 
-import           Cardano.BM.Data.LogItem (LogObject(..))
-import           Cardano.BM.Data.Tracer (Tracer(..))
+import           Cardano.BM.Data.LogItem (LogObject(..), LoggerName)
+import           Control.Tracer
 
 \end{code}
 %endif
 
 \subsubsection{Trace}\label{code:Trace}\index{Trace}
-A |Trace m a| is a |Tracer m (LogObject a)|.
+A |Trace m a| is a |Tracer| containing the context name and a |LogObject a|.
 \begin{code}
 
-type Trace m a = Tracer m (LogObject a)
+type Trace m a = Tracer m (LoggerName, LogObject a)
```

Comment:
Maybe define a newtype wrapper for `(LoggerName, LogObject a)`.

Reply:
I tried with `newtype` but keeping it as an alias is more practical in this case. Some functions accept a `Tracer m a` and the aliased `Trace m a` still is compatible with the more generic type.